From foo@baz Wed Jul 29 15:42:27 PDT 2009
Date: Wed, 29 Jul 2009 15:42:27 -0700
To: Greg KH <greg@kroah.com>
From: Greg Kroah-Hartman <gregkh@suse.de>
Subject: Staging: hv: remove duplicated osd.o inclusions

From: Greg Kroah-Hartman <gregkh@suse.de>

Now that we have properly prefixed the osd.c functions, we don't need to
include it in each of the modules.  So only build it into the hv_vmbus
module.

Export the symbols that the other hv_ modules need, so that they can
properly find them.

Cc: Bill Pemberton <wfp5p@virginia.edu>
Cc: Hank Janssen <hjanssen@microsoft.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>
---
 drivers/staging/hv/Makefile |    6 +++---
 drivers/staging/hv/osd.c    |    6 ++++++
 2 files changed, 9 insertions(+), 3 deletions(-)

--- a/drivers/staging/hv/Makefile
+++ b/drivers/staging/hv/Makefile
@@ -4,6 +4,6 @@ obj-$(CONFIG_HYPERV_BLOCK)	+= hv_blkvsc.
 obj-$(CONFIG_HYPERV_NET)	+= hv_netvsc.o
 
 hv_vmbus-objs := vmbus_drv.o osd.o Sources.o
-hv_storvsc-objs := storvsc_drv.o osd.o StorVsc.o
-hv_blkvsc-objs := blkvsc_drv.o osd.o BlkVsc.o
-hv_netvsc-objs := netvsc_drv.o osd.o NetVsc.o RndisFilter.o
+hv_storvsc-objs := storvsc_drv.o StorVsc.o
+hv_blkvsc-objs := blkvsc_drv.o BlkVsc.o
+hv_netvsc-objs := netvsc_drv.o NetVsc.o RndisFilter.o
--- a/drivers/staging/hv/osd.c
+++ b/drivers/staging/hv/osd.c
@@ -79,6 +79,7 @@ void *osd_PageAlloc(unsigned int count)
 	/* if (p) memset(p, 0, PAGE_SIZE); */
 	/* return p; */
 }
+EXPORT_SYMBOL_GPL(osd_PageAlloc);
 
 void osd_PageFree(void* page, unsigned int count)
 {
@@ -86,6 +87,7 @@ void osd_PageFree(void* page, unsigned i
 	/*struct page* p = virt_to_page(page);
 	__free_page(p);*/
 }
+EXPORT_SYMBOL_GPL(osd_PageFree);
 
 struct osd_waitevent *osd_WaitEventCreate(void)
 {
@@ -99,12 +101,14 @@ struct osd_waitevent *osd_WaitEventCreat
 	init_waitqueue_head(&wait->event);
 	return wait;
 }
+EXPORT_SYMBOL_GPL(osd_WaitEventCreate);
 
 void osd_WaitEventSet(struct osd_waitevent *waitEvent)
 {
 	waitEvent->condition = 1;
 	wake_up_interruptible(&waitEvent->event);
 }
+EXPORT_SYMBOL_GPL(osd_WaitEventSet);
 
 int osd_WaitEventWait(struct osd_waitevent *waitEvent)
 {
@@ -115,6 +119,7 @@ int osd_WaitEventWait(struct osd_waiteve
 	waitEvent->condition = 0;
 	return ret;
 }
+EXPORT_SYMBOL_GPL(osd_WaitEventWait);
 
 int osd_WaitEventWaitEx(struct osd_waitevent *waitEvent, u32 TimeoutInMs)
 {
@@ -126,6 +131,7 @@ int osd_WaitEventWaitEx(struct osd_waite
 	waitEvent->condition = 0;
 	return ret;
 }
+EXPORT_SYMBOL_GPL(osd_WaitEventWaitEx);
 
 static void osd_callback_work(struct work_struct *work)
 {
